tests/test_app/TestApp/Controller/PostsController.php

```diff
@@ -32,6 +32,16 @@ public function initialize(): void
         $this->loadComponent('Flash');
         $this->loadComponent('RequestHandler');
         $this->loadComponent('FormProtection');
+
+        $this->middleware(function ($request, $handler) {
```

Comment:
This can place nice with the DI container as middleware could be constructor parameters too :+1:

Reply:
Yeah the controller factory has the reference to the container instance so it opens up possibilities.